cmd/bridge/main.go

```diff
@@ -121,20 +122,36 @@ func main() {
 
 	fLoadTestFactor := fs.Int("load-test-factor", 0, "DEV ONLY. The factor used to multiply k8s API list responses for load testing purposes.")
 
-	if err := fs.Parse(os.Args[1:]); err != nil {
+	fs.String("developer-catalog-categories", "", "Allow catalog categories customization. (JSON as string)")
+
+	// Read configuration from
+	// 1. Config file
+	// 2. Environment variables (overrides config file)
+	// 3. Commandline arguments (overrides config file and environment varibles)
+	//
+	// Because the config filename could be defined as commandline argument or
+	// environment variable, we need to parse these inputs before reading the
+	// config file and need to override the config values after this again.
```

Comment:
We should add some tests to make sure we're really honoring this and to avoid regressions.

Reply:
Extract flag set parsing into serverconfig package, please take a look at main.go, config.go and config_test.go :)